src/hooks/historikk/useDialogMedBehandlerHistorikk.ts

```diff
@@ -0,0 +1,59 @@
+import { HistorikkEvent } from "@/data/historikk/types/historikkTypes";
+import { useBehandlerdialogQuery } from "@/data/behandlerdialog/behandlerdialogQueryHooks";
+import {
+  MeldingDTO,
+  MeldingResponseDTO,
+} from "@/data/behandlerdialog/behandlerdialogTypes";
+import { meldingTypeTexts } from "@/data/behandlerdialog/behandlerdialogTexts";
+
+interface DialogMedBehandlerHistorikk {
+  isDialogMedBehandlerHistorikkLoading: boolean;
+  isDialogMedBehandlerHistorikkError: boolean;
+  dialogMedBehandlerHistorikk: HistorikkEvent[];
+}
+
+function beregnAvsender(dialogmelding: MeldingDTO): string {
+  return dialogmelding.innkommende
+    ? dialogmelding.behandlerNavn ?? "Mangler navn på behandler"
+    : dialogmelding.veilederIdent ?? "Mangler navn på veileder";
+}
+
+function createHistorikkEventsFromDialogMedBehandler(
```

Comment:
Siden vi allerede er i kontekst av dialog med behandler holder det kanskje med dette? 😊
```suggestion
function createHistorikkEvents(
```